curation-migration-datasync/eventConsumer.ts

```diff
@@ -0,0 +1,123 @@
+import { DataService } from './database/dataService';
+import { getTopicForReaditLaTmpDatabase } from './helpers/topicMapper';
+import { getParserMetadata } from './externalCaller/parser';
+import { AddScheduledItemPayload, TileSource } from './types';
+import {
+  hydrateCuratedFeedItem,
+  hydrateCuratedFeedProspectItem,
+  hydrateCuratedFeedQueuedItem,
+} from './helpers/hydrator';
+import { CuratedItemRecord, ScheduledSurfaceGuid } from './dynamodb/types';
+import { insertCuratedItem } from './dynamodb/curatedItemIdMapper';
+import { dbClient } from './dynamodb/dynamoDbClient';
+import { Knex } from 'knex';
+
+/**
+ * function to generate epoc timestamp
+ * @param scheduledDate
+ */
+export function convertDateToTimestamp(scheduledDate: string) {
+  return Math.round(new Date(scheduledDate).getTime() / 1000);
+}
+
+/**
+ * fetches `cohara` from 'ad|Mozilla-LDAP|cohara'
+ * @param ssoName
+ */
+export function getCuratorNameFromSso(ssoName: string) {
+  const prefix = 'ad|Mozilla-LDAP|';
+  if (!ssoName.startsWith(prefix)) {
+    throw new Error(
+      'unexpected sso format, createdBy are expected to startWith `ad|Mozilla-LDAP|`'
+    );
+  }
+  return ssoName.substring(prefix.length);
+}
+
+/**
+ * converts the event body of `add-scheduled-item` event to database models,
+ * and inserts them to the database within one transaction. If successful,
+ * adds the curatedRecId and externalIds from the event to the dynamoDb
+ * @param eventBody
+ */
+export async function addScheduledItem(
+  eventBody: AddScheduledItemPayload,
+  db: Knex
+) {
+  let curatedRecId = -1;
+  const curatedItemService = new DataService(db);
+
+  const topicId = await curatedItemService.getTopicIdByName(
+    getTopicForReaditLaTmpDatabase(eventBody.topic)
+  );
+
+  const parserResponse = await getParserMetadata(eventBody.url);
```

Comment:
did we implement a back-off strategy here? We called it out that the parser might intermittently fail and the request would need to be retried. This could also be in a follow-up ticket given the size of this PR.

Reply:
ya can handle this in follow up

Reply:
oh, yeah given that this isn't going to be retried automatically since it's not SQS + lambda?